### DISCUSSION TITLE
Commands git

### DISCUSSION THREAD

**Discussion by alcdevel** [community/community#60993] [New to GitHub] [answered]
Hey Im new to github and I wanted to know the basics commands of git to use on github

**Comment by Gustavonn07** [comment, marked as answer]
Hey this is the basics git commands I know!
The most useful (for me) is push, add, commit, pull and status

1. Assign user to active session
`git config --global user.name "<your_name>"`

2. Assign email to active session:
`git config --global user.email "<your_email>"`

3. Create a new branch or branch:
`git checkout -b <branch name>`

4. Change branch/branch:
`git checkout <branch name>`

5. Initialize the ".git/" file to control the folder on the local machine that will later upload to the Github repository:
`git init`

6. Clone/download the entire project EXCLUSIVELY from YOUR remote repository, informing the link of the repository where the fork was given and not the original project:
`git clone <url>`

7. Validate modified files within the project. In red it shows the modified files. In green, it shows the files that were added by "git add":
`git status`

8. Placing the modified file in a safe area (The "." after the 'git add' command adds all files in red indicated after the git status):
`git add .`

9. Create a new version of the project with the creator references:
`git commit -m "<change_text>"`

10. Change the main branch name from master to main (this is currently recommended best practice)
`git branch -M "main"`

11. Transferring the commit from the local repository (from my machine) to a repository on the Github platform (Origin is the name used to refer to our repository. Once this is done, the local repository will be connected to the Github repository, however, the commit we give on the machine does not automatically go up to the platform...)
`git remote add origin <repository link>`

12. Upload, for the first time, the changes to the remote repository (github):
`git push -u origin <branch you want to push (main or develop, for example)>`

13. Up, after the first time (without the "-u", the changes to the remote repository (github)):
`git push origin <branch you want to push (main or develop, for example)>`

14. Pull changes from repository:
`git pull`

15. Validate my comments and modifications:
`git log`

16. Add the contents of the current branch to the contents of another branch:
`git merge <branch_name>`

17. Show the changes made:
`gitk`

18. To check the installed git version
`git -v`

Hope it helps!

**Reply by alcdevel** [reply]
Wow thanksss for being this fast it will really help me (I was just ctrl-c ctrl-v my codes...)
